imap_processing/lo/l0/data_classes/science_direct_events.py

```diff
@@ -0,0 +1,213 @@
+    decompression of L0 to L1A data.
+
+
+    The TOF data in the binary is in the following order:
+    ABSENT, TIME, ENERGY, MODE, TOF0, TOF1, TOF2, TOF3, CKSM, POS
+
+    ABSENT, TIME, ENERGY, and MODE will be present for every type of DE.
+
+    ABSENT: signals the case number for the DE (4 bits).
+    TIME: the time of the DE (12 bits).
+    ENERGY: Energy step (3 bits).
+    MODE: Signals how the data is packed. If MODE is 1, then the TOF1
+    (for case 1a) will need to be calculated using the checksum and other TOFs
+    in the L1B data product.
+    If MODE is 0, then there was no compression and all TOFs are transmitted.
+
+    The presence of TOF0, TOF1, TOF2, TOF3, CKSM, and POS depend on the
+    case number.
+
+    - Case 0 can either be a gold or silver triple. Gold triples do
+    not send down the TOF1 value and instead recover the TOF1 value
+    on the ground using the decompressed checksum.
+    - Cases 4, 6, 10, 12, 13 may be Bronze. If it's not a bronze,
+    the Position is not transmitted, but TOF3 is. If it is bronze, the table
+    should be used as is. If it's not bronze, position was not transmitted,
+    but TOF3 was transmitted.
+    - Cases 1, 2, 3, 5, 7, 9, 13 will always have a MODE of 0, so the same
+    fields will always be transmitted.
+
+    Bit Shifting:
+    TOF0, TOF1, TOF2, TOF3, and CKSM all must be shifted by one bit to the
+    left. All other fields do not need to be bit shifted.
+
+    The raw values are computed for L1A and will be converted to
+    engineering units in L1B.
+
+    Attributes
+    ----------
+    SHCOARSE : int
+        Spacecraft time.
+    COUNT: int
+        Number of direct events.
+    DATA: str
+        Compressed TOF Direct Event time tagged data.
+    TIME: numpy.ndarray
+        time tag for the direct event
+    ENERGY: numpy.ndarray
+        energy of the direct event ENA.
+    MODE: numpy.ndarray
+        Indication of how the data is packed.
+    TOF0: numpy.ndarray
+        Time of Flight 0 value for direct event.
+    TOF1: numpy.ndarray
+        Time of Flight 1 value for direct event.
+    TOF2: numpy.ndarray
+        Time of Flight 2 value for direct event.
+    TOF3: numpy.ndarray
+        Time of Flight 3 value for direct event.
+    CKSM: numpy.ndarray
+        This is checksum defined relative to the TOFs
+        condition for golden triples. If golden triples are below
+        a certain threshold in checksum it's considered golden, otherwise,
+        it's considered a silver triple. This is important for the compression
+        for golden triples because it's used to recover TOF1 because
+        compression scheme to save space on golden triples doesn't send
+        down TOF1 so it's recovered on the ground using the checksum
+    POS: numpy.ndarray
+        Stop position for the direct event. There are 4 quadrants
+        on the at the stop position.
+
+    Methods
+    -------
+    __init__(packet, software_vesion, packet_file_name):
```

Comment:
can you add a docstring about `packet` parameter? I think it's a list of decom packets, is that right?

Reply:
That's correct. Yes, I'll add that to the __init__ docstring